### PR TITLE
Clock Init

### PR DESCRIPTION
**Pull request by rjoleary**

### CODE REVIEW

File: src/soc/sifive/fu540/src/clock.rs [hunk 1]
```diff
@@ -376,59 +275,37 @@ impl Clock {
...
-        // 	// get all DDR out of reset
-        // 	write32(&prci->devicesresetreg,
-        // 		PRCI_DEVICESRESET_DDR_CTRL_RST_N(1) |
-        // 		PRCI_DEVICESRESET_DDR_AXI_RST_N(1) |
-        // 		PRCI_DEVICESRESET_DDR_AHB_RST_N(1) |
-        // 		PRCI_DEVICESRESET_DDR_PHY_RST_N(1));
-
-        // 	// HACK to get the '1 full controller clock cycle'.
+
+        // HACK to get the '1 full controller clock cycle'.
```
Comment (rminnich) [inline src/soc/sifive/fu540/src/clock.rs]: I think this comment should stay in, it is required.

File: src/soc/sifive/fu540/src/clock.rs [hunk 2]
```diff
@@ -376,59 +275,37 @@ impl Clock {
...
         architecture::fence();
```
Comment (rminnich) [inline src/soc/sifive/fu540/src/clock.rs]: you can change it to 
// required delay to get '1 full controller clock cycle'

File: src/soc/sifive/fu540/src/clock.rs [hunk 3]
```diff
@@ -376,59 +275,37 @@ impl Clock {
...
+
+        architecture::fence();
     }
 }
 
 // NASTY. We'll live with it for now.
 const UART_DEVICES: u32 = 2;
 const UART_REG_DIV: u32 = 0x18;
-const UART_DIV_VAL: u32 = 4;
+const UART_DIV_VAL: u32 = 4340;
```
Comment (rminnich) [inline src/soc/sifive/fu540/src/clock.rs]: I wonder if due to other bugs coreboot never got this right ...

File: src/soc/sifive/fu540/src/clock.rs [hunk 4]
```diff
@@ -444,347 +321,8 @@ fn update_peripheral_clock_dividers() {
...
-/* DDR PHY reset (active low) */
-/* 4 */
-/* Reserved */
-/* 5 */
-/* GEMGXL_RST_N */
-/* RW */
-/* 0x0 */
-/* Gigabit Ethernet Subsystem reset (active low) */
+// TODO: There might be a better way to detect whether we are running in QEMU.
+fn is_qemu() -> bool {
```
Comment (rminnich) [inline src/soc/sifive/fu540/src/clock.rs]: NICE!

File: src/soc/sifive/fu540/src/clock.rs (hunk 3 above)
Comment (rjoleary) [reply src/soc/sifive/fu540/src/clock.rs]: Yes, 4 is definitely wrong. The datasheet even says the value cannot be less than 16.

**Pull request merged**